layers/core_validation.cpp

```diff
@@ -1603,30 +1603,29 @@ bool CoreChecks::ValidateCmdBufDrawState(const CMD_BUFFER_STATE &cb_state, CMD_T
 
     bool result = false;
 
-    if (pipeline.descriptor_buffer_mode) {
-        if (!last_bound.per_set.empty()) {
-            bool non_push = false;
-            for (const auto &ds : last_bound.per_set) {
-                non_push = ds.bound_descriptor_set && !ds.bound_descriptor_set->IsPushDescriptor();
-
-                if (non_push) break;
+    for (uint32_t i = 0; i < last_bound.per_set.size(); i++) {
+        const auto &ds = last_bound.per_set[i];
```

Comment:
range-based for loop

```suggestion
    for (const auto &ds : last_bound.per_set) {
```